pkg/resources/instance/client.go

```diff
@@ -210,20 +227,81 @@ func (r *StatusClient) GetPgControlDataFromInstance(
 	return result.Data, result.Error
 }
 
+// UpgradeInstanceManager upgrades the instance manager to the passed availableArchitecture
+func (r *StatusClient) UpgradeInstanceManager(
+	ctx context.Context,
+	pod *corev1.Pod,
+	availableArchitecture *utils.AvailableArchitecture,
+) error {
+	contextLogger := log.FromContext(ctx)
+
+	binaryFileStream, err := availableArchitecture.FileStream()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if binaryErr := binaryFileStream.Close(); binaryErr != nil {
+			contextLogger.Error(err, "while closing the binaryFileStream")
+		}
+	}()
+
+	scheme := GetStatusSchemeFromPod(pod)
+	updateURL := url.Build(scheme, pod.Status.PodIP, url.PathUpdate, url.StatusPort)
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, updateURL, nil)
+	if err != nil {
+		return err
+	}
+	req.Body = binaryFileStream
+
+	resp, err := r.Client.Do(req)
+	// This is the desired response. The instance manager will
+	// synchronously update and this call won't return.
+	if isEOF(err) {
```

Comment:
I still think that is weird that the expected function outcome is not at the end, probably I will refactor this further